### PR TITLE
Adding Repo Management for Centos, Redhat, LinuzAmazon

### PR DESCRIPTION
**Pull request by alexvaque**
Adding Repo Management for Centos, Redhat, LinuzAmazon

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -52,4 +51,11 @@
     scheduler => {},
     mysql_replication_hostgroups => {},
   }
+
+  # 'CentOS', 'Fedora', 'Scientific', 'RedHat', 'Amazon', 'OracleLinux'
+  $rpm_repo_name   = 'mariadb_repo'
+  $rpm_repo_descr  = 'mariadb_repo_for_proxysql'
+  $rpm_repo        = 'http://yum.mariadb.org/10.1/centos6-amd64'
```
Comment (mcrauwel) [inline manifests/params.pp]: is this the correct repo url?

**Comment by bastelfreak** [comment]
Hi @alexvaque, please check the used email address in your commit. It is currently not associated with your github account.

**Comment by alexvaque** [comment]
Yes, it is correct REPO URL .

 

We could use MARIADB_REPO or Percona. Personally, I prefered to use MARIADB_REPO , Percona REPO runs too.

 


> yum install http://www.percona.com/downloads/percona-release/redhat/0.1-4/percona-release-0.1-4.noarch.rpm



Enviar: miércoles 8 de marzo de 2017 a las 18:48
De: "Matthias Crauwels" <notifications@github.com>
Para: voxpupuli/puppet-proxysql <puppet-proxysql@noreply.github.com>
CC: alexvaque <alex.vaque@gmx.com>, Author <author@noreply.github.com>
Asunto: Re: [voxpupuli/puppet-proxysql] Adding Repo Management for Centos, Redhat, LinuzAmazon (#16)


@mcrauwel commented on this pull request.


In manifests/params.pp:

> @@ -52,4 +51,11 @@
     scheduler => {},
     mysql_replication_hostgroups => {},
   }
+
+  # 'CentOS', 'Fedora', 'Scientific', 'RedHat', 'Amazon', 'OracleLinux'
+  $rpm_repo_name   = 'mariadb_repo'
+  $rpm_repo_descr  = 'mariadb_repo_for_proxysql'
+  $rpm_repo        = 'http://yum.mariadb.org/10.1/centos6-amd64'


is this the correct repo url?

**Comment by mcrauwel** [comment]
but in neither of those repo's there is a proxysql package...

**Comment by alexvaque** [comment]
Hello McRauwel

About that.
> but in neither of those repo's there is a proxysql package...
I confirm that If you install the Percona or Mariadb Repository , you can find a proxysql package.

For example you could check it doing :
# yum install http://www.percona.com/downloads/percona-release/redhat/0.1-4/percona-release-0.1-4.noarch.rpm 
# yum install proxysql

Actually, I am thinking that maybe will be better to install through the RPM directly without repository update. 

Doing something like that, but using Puppet. I need to think how to adapt this idea in your Puppet Module
> yum install https://www.percona.com/redir/downloads/proxysql/proxysql-1.3.2/binary/redhat/6/x86_64/proxysql-1.3.2-1.1.x86_64.rpm

What do you think?

I am checking it and I will add spec tests and add the supported OS's to metadata.json that you mentioned me . 


Thank you

**Comment by mcrauwel** [comment]
@alexvaque that looks good! Percona seems to offer a package manager for proxysql (just not the latest releases)

also look at the travis output, you are mixing spaces and tabs. Travis (and rubocop) don't like that!

thnx in advance!

**Comment by bastelfreak** [comment]
rubocop provides a nice way to fix errors, works often:
```bash
bundle exec rubocop -c .rubocop.yml -D -S -a
```

**Comment by mcrauwel** [comment]
it look nice, I will try to test tonight, could you still look at the metadata.json file so the new OS's get tested also?

**Comment by alexvaque** [comment]
>  Could you still look at the metadata.json file so the new OS's get tested also?
OK , I will check

You can see that now the module can manage the Proxysql through Package RPM or through Repository

**through Repository**

include proxysql (by default using Package) 

Also we can change the repository with something like that: 
  class { 'proxysql':
    rpm_repo_name   => 'percona_repo',
    rpm_repo_descr  => 'percona_repo_contains_proxysql',
    rpm_repo        => 'http://repo.percona.com/release/$releasever/RPMS/$basearch',
    rpm_repo_key    => 'https://www.percona.com/downloads/RPM-GPG-KEY-percona',
  }

**through Package**
If someone prefer to use RPM Package we can use the following

Using the default RPM added in the module
  class { 'proxysql':
    manage_rpm               => true,
  }  

Or using specific RPM URL

  class { 'proxysql':
    manage_rpm               => true,
    package_source           => 'https://www.percona.com/redir/downloads/proxysql/proxysql-1.3.2/binary/redhat/6/x86_64/proxysql-1.3.2-1.1.x86_64.rpm',
  }

**Comment by alexvaque** [comment]
Good morning , Any news about the last changes? Do you need some improvements or changes? Or last changes are OK ? Thank you :)

**Comment by mcrauwel** [comment]
I'm just updating the readme and docs so we can start publishing this to the forge also! thank you for your contribution!

**Pull request merged**